.github/workflows/cleanup.yml

```diff
@@ -0,0 +1,140 @@
+## Github workflow to clean up old caches and workflow runs
+##   TODO: look into using  https://github.com/actions/github-script
```

Comment:
Would we do this to break the scripts up into smaller script steps?

Reply:
possibly - it may make the logic a lot easier to follow as well (as evidenced by your other comment). 
using the shell/gh cli it works...but as you noticed it's pretty ugly